growattServer/__init__.py

```diff
@@ -116,12 +116,15 @@ def login(self, username, password, is_password_hashed=False):
             'userName': username,
             'password': password
         })
-        data = json.loads(response.content.decode('utf-8'))['back']
-        if data['success']:
-            data.update({
-                'userId': data['user']['id'],
-                'userLevel': data['user']['rightlevel']
-            })
+        try:
+            data = json.loads(response.content.decode('utf-8'))['back']
+            if data['success']:
+                data.update({
+                    'userId': data['user']['id'],
+                    'userLevel': data['user']['rightlevel']
+                })
+        except:
+            data = {'succsss': False}
```

Comment:
There's a typo here in the parameter `success`

Reply:
Good spot! Now I understand why my bits in Node-Red weren't working as they should. :)